applications/DEMApplication/custom_utilities/post_utilities.h

```diff
@@ -96,41 +96,28 @@ namespace Kratos {
 
             ElementsArrayType& pElements = rModelPart.GetCommunicator().LocalMesh().Elements();
 
-            OpenMPUtils::CreatePartition(ParallelUtilities::GetNumThreads(), pElements.size(), this->GetElementPartition());
-
             double velocity_X = 0.0, velocity_Y = 0.0, velocity_Z = 0.0;
-
             int number_of_elements = 0;
-
             #pragma omp parallel for reduction(+: velocity_X, velocity_Y, velocity_Z, number_of_elements)
```

Comment:
same

Reply:
@philbucher I have a doubt with this one. 
#pragma omp parallel for reduction(+: velocity_X, velocity_Y, velocity_Z, number_of_elements)
should I do a separate block_for_each<MaxReduction for each one or is there a way to put them all together?

thx!

Reply:
nvm, just found about block_for_each<MultipleReduction>. Going to try that.

Reply:
another option would be to use a `std::tuple<double, double, double, std::size_t>`

Reply:
@philbucher is it possible to break out of a block_for_each loop with a continue or a break? 
similar to:
block_for_each(pElements, [&](ModelPart::ElementType& rElement) {
            if (rElement.Is(BLOCKED)) continue;
            SphericParticle& spheric_particle = dynamic_cast<SphericParticle&>(rElement);
            if (!(*(spheric_particle.mpInlet))[DENSE_INLET]) continue;

Reply:
afaik openmp doesn't support either

since we call a lambda for each index you could try with `return` instead of `continue`, I think it should achieve the same behavior

can you test and try please?

Reply:
According to std:for_each documentation, using return instead of continue in this case is ok. The real problem appears when you have to deal with continue and break in the same loop (requiring to create pseudo-functions for both).